### PR TITLE
Now passing strings to mimetypes.guess_extension.

### LINKED ISSUE #552: cachedataset extension derivation broken
Consider this code:

```
import pyvo

ACCESS_URL = "http://dc.g-vo.org/maidanak/res/rawframes/siap/siap.xml?"

svc = pyvo.sia.SIAService(ACCESS_URL)
images = svc.search((340.1,3.36), size=(0.1, 0.1))
images[0].cachedataset()
```

At this moment, this will result in a file Maidanak_Q2237+0305_2004-08-24_00:00:00_Johnson_R-1.None (or somesuch) on disk; the extension .None is really suboptimal.

The reason extension inference fails is that dal.mimetype.mime2extension, probably for reasons lost in time, turns the media type to a bytestring before passing it on to the built-in module mimetypes.  Passing in a string fixes that and yields a .fits (on systems with suitable mime.types), as expected.

### PR DESCRIPTION
**Pull request by msdemlei**
Also changing the default for images to .img (from None).

This is supposed to address bug #552.

Fixes #552

### CODE REVIEW

**Comment by msdemlei** [comment]
So, I think the remaining failure is unrelated (a fluke?).  From my point of view, this is ready to go.

File: pyvo/dal/sia.py [hunk 1]
```diff
@@ -903,7 +903,7 @@ def suggest_dataset_basename(self):
             out = re.sub(r'\s+', '_', out.strip())
         return out
 
-    def suggest_extension(self, *, default=None):
+    def suggest_extension(self, *, default='img'):
```
Comment (andamian) [inline pyvo/dal/sia.py:906]: Just curious why `img`? I've noticed same assumption in `suggest_dataset_basename`.

**Comment by msdemlei** [comment]
On Fri, Jun 14, 2024 at 03:12:32PM -0700, Adrian wrote:
> @andamian commented on this pull request.
>
>
>
> > @@ -903,7 +903,7 @@ def suggest_dataset_basename(self):
>              out = re.sub(r'\s+', '_', out.strip())
>          return out
>
> -    def suggest_extension(self, *, default=None):
> +    def suggest_extension(self, *, default='img'):
>
> Just curious why `img`? I've noticed same assumption in
> `suggest_dataset_basename`.

I'm not particularly married to "img", and now that you mention it,
since this is not only for SIAP (and hence images), I notice it may
if fact not be particularly appropriate.  I certainly have no strong
preferences, except the None we have now seems a choice that will
confuse our users more than just about anything else.  What about
"dat"?  "bin"?  "octets"?  "data"?

**Comment by msdemlei** [comment]
There is "dat" as a non-informative default extension elsewhere in pyvo, so that's what I'm going for now.

**Comment by msdemlei** [comment]
(test failures were a glitch on a remote service)

File: pyvo/dal/adhoc.py [hunk 2]
```diff
@@ -629,6 +635,15 @@ def get_first_proc(self):
             return proc
         raise IndexError("No processing service found in datalink result")
 
+    def set_original_row(self, row):
+        """
+        note the table row this DatalinkResults instance was generated
+        from.
+
+        This is mostly for pyVO-internal use.
```
Comment (bsipocz) [inline pyvo/dal/adhoc.py:643]: given this, should this be prepended with an `_`?

**Review by bsipocz — COMMENTED** [review]
Can we have a test for `original_row`, and its setter method? As far as I see it doesn't come up with any of the tests.

**Comment by bsipocz** [comment]
OK, looking at the other PRs and discussions, and the commit history here, I think that the `original_row` related changes are added here by mistake in the very last commit iteration. 

So, I'll make a backup of the current branch, but then will go ahead and remove those changes from this PR and merge it as the rest are fully finished with test and addressed review.

**Comment by bsipocz** [comment]
The backup commit is here: https://github.com/astropy/pyvo/commit/e2cdd91942f2516590fbd11c8f649cca6fb5b094

**Review by bsipocz — APPROVED** [review]

**Comment by bsipocz** [comment]
Thank you @msdemlei!

(I mark this for backport, in case we have 1.5.3 coming out sooner than 1.6)

**Comment by msdemlei** [comment]
On Thu, Jun 27, 2024 at 01:59:33PM -0700, Brigitta Sipőcz wrote:
> @bsipocz commented on this pull request.
>
> Can we have a test for `original_row`, and its setter method? As
> far as I see it doesn't come up with any of the tests.

Sorry that that code appeared in this PR in the first place -- I was
juggling a few too many branches at that moment.  In the actual PR
that this stuff belongs to, #559, the setter method is (fortunately)
gone, and there are some basic tests for the main code paths
involving original_row, too.

**Pull request merged**